### PR TITLE
Feature: MasterAwareV2 normalization

### PR DESCRIPTION
**Pull request by danoctavian**
## Context

Issue: https://github.com/NexusMutual/smart-contracts/issues/382

Full motivation for the change is explained in this doc:
https://docs.google.com/document/d/1s-L1FvBelHK-T009ydJL2R0KYRSZapgfbcq4IKqbN1E/edit?usp=sharing

Initially we had the Iupgradable abstract contract that was poorly named and had duplicate values in storage (ms and master). That was renamed to LegacyMasterAware, but nothing was changed in it.

Secondly we have MasterAware which provides basically the same functionality but has no storage duplicates and has sane modifier names.

Thirdly we have MasterAwareV2 which is the new interface that uses a mapping from an Id to a contract address to store the dependencies. This is nicer because it takes only 2 memory slots (the mapping and the master address).

## Changes proposed in this pull request

We want to normalize the use of MasterAwareV2 as much as possible.

Here we are moving MCR, MemberRoles and TokenController to MasterAwareV2 interface.

It works because you can drop in replace LegacyMasterAware with MasterAwareV2 - they both take 2 memory slots AND the mapping ```  mapping(uint => address payable) internal internalContracts;``` from MasterAwareV2 is falling in the place of ```  INXMMaster public ms;```. 

This is considered safe because the mapping memory slot does not need to be cleared - it is not used by the mapping data structure.

Motivation for skipping ProposalCategory and Governance:

They are to be refactored when we are overhauling governance fully. The benefits are not there for this upgrade right now.

Hurdles: Governance.sol is fixed to 0.5.0 and needs updates to comply with 0.8.16. 

ProposalCategory generates a stack to deep error when upgraded to the 0.8.16 compiler.

## Test plan

No extra tests added. All current tests required to pass.

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: contracts/modules/governance/MemberRoles.sol [hunk 1]
```diff
@@ -223,14 +233,16 @@ contract MemberRoles is IMemberRoles, Governed, LegacyMasterAware {
...
   ) external override {
     _switchMembership(msg.sender, newAddress);
-    nxm.transferFrom(msg.sender, newAddress, nxm.balanceOf(msg.sender));
+    INXMToken _token = token();
+    _token.transferFrom(msg.sender, newAddress, _token.balanceOf(msg.sender));
 
+    ICover _cover = cover();
     // Transfer the cover NFTs to the new address, if any were given
-    cover.transferCovers(msg.sender, newAddress, coverIds);
+    cover().transferCovers(msg.sender, newAddress, coverIds);
```
Comment (kyledewy) [inline contracts/modules/governance/MemberRoles.sol:241]: `cover()` is called twice here. i think you meant to use the `_cover` variable instead?

File: contracts/interfaces/IMasterAwareV2.sol [hunk 2]
```diff
@@ -10,7 +10,10 @@ interface IMasterAwareV2 {
     MR, // MemberRoles.sol
     MC, // MCR.sol
     CO, // Cover.sol
-    AS  // Assessment.sol
+    AS, // Assessment.sol
```
Comment (kyledewy) [inline contracts/interfaces/IMasterAwareV2.sol:13]: [TK, PS, GV] are missing comments for which contract they reference.

**Review by kyledewy — APPROVED** [review]
Left a couple minor comments, but otherwise LGTM :+1:

**Review by shark0der — CHANGES_REQUESTED** [review]
Let's make an effort to get rid of `MasterAware.sol`

**Comment by danoctavian** [comment]
Removed MasterAware from all except LegacyGateway.sol

As discussed with @roxdanila  we'll postpone messing with that contract until we clarify the biz requirements around what we need to support there.

Then we'll cleanup the MasterAware as well.

File: contracts/modules/legacy/LegacyPooledStaking.sol [hunk 3]
```diff
@@ -44,10 +44,7 @@ contract LegacyPooledStaking is IPooledStaking, MasterAware {
 
   /* Storage variables */
 
-  bool public initialized;
-
-  INXMToken public token;
-  ITokenController public tokenController;
+  ITokenController public _unused0;
```
Comment (shark0der) [inline contracts/modules/legacy/LegacyPooledStaking.sol:47]: `token` and `initialized` were removed - let's leave a comment of what happens with the storage layout here as it's not obvious why this is OK

File: contracts/modules/capital/Pool.sol [hunk 4]
```diff
@@ -405,14 +401,29 @@ contract Pool is IPool, MasterAware, ReentrancyGuard {
...
    * @dev Implements MasterAware interface function
    */
   function changeDependentContractAddress() public {
-    nxmToken = INXMToken(master.tokenAddress());
-    tokenController = ITokenController(master.getLatestAddress("TC"));
-    mcr = IMCR(master.getLatestAddress("MC"));
+    internalContracts[uint(ID.TK)] = payable(master.tokenAddress());
+    internalContracts[uint(ID.TC)] = master.getLatestAddress("TC");
+    internalContracts[uint(ID.MC)] = master.getLatestAddress("MC");
+    internalContracts[uint(ID.MR)] = master.getLatestAddress("MR");
```
[the remark this answers is not in the document]
Comment (danoctavian) [reply contracts/modules/capital/Pool.sol:426]: Yes. 

Any MasterAwareV2 contract is using it:

https://github.com/NexusMutual/smart-contracts/blob/3bced02d04b378ed8e70deaadd1ef33a2112fb39/contracts/abstract/MasterAwareV2.sol#L17

**Review by shark0der — APPROVED** [review]

**Pull request merged**